packages/services/src/assets/stacks-asset.utils.ts

```diff
@@ -85,8 +87,39 @@ export function createSip10CryptoAssetInfo(
     contractId: getContractPrincipalFromAssetIdentifier(assetIdentifier),
     decimals: metadata.decimals ?? 0,
     hasMemo: isTransferableSip10Token(metadata),
-    imageCanonicalUri: metadata.image_canonical_uri ?? '',
+    imageCanonicalUri: createSip10ImageCanonicalUri({ metadata, assetName }),
     name,
     symbol: metadata.symbol || getTicker(name),
   };
 }
+
+interface Sip10ImageCanonicalUriArgs {
+  metadata: HiroFtMetadataResponse;
+  assetName: string;
+}
+export function createSip10ImageCanonicalUri({
+  metadata,
+  assetName,
+}: Sip10ImageCanonicalUriArgs): string {
+  if (isUndefined(metadata.image_canonical_uri) || metadata.image_canonical_uri === '') {
+    const tokenInfoFromAlexLabs = alexTokenData.find(({ name }) => assetName === name);
+
+    console.log(
+      'createSip10ImageCanonicalUri metadata',
+      metadata.image_canonical_uri,
+      assetName,
+      tokenInfoFromAlexLabs?.icon
+    );
+    return tokenInfoFromAlexLabs?.icon ?? '';
+  }
+  return metadata.image_canonical_uri;
+}
+// TODO: use this to fetch data from alex lab once determined where to call it from
```

Comment:
@alexp3y : I was working on adding this to patch in where we are missing some token URIs. Sometimes the `HiroFtMetadataResponse` contains empty / undefined `metadata.image_canonical_uri`.

Can you recommend a sensible place in the service to:
- call `fetchDataFromAlexLab` and fetch their JSON once 
- cache the result in a `const` 
- use it as a lookup object in `createSip10CryptoAssetInfo` to add in missing avatars